nixos/lib/test-driver/lib/driver.py

```diff
@@ -0,0 +1,256 @@
+from contextlib import contextmanager
+from queue import Queue, Empty
+from typing import Tuple, Any, Dict, Iterator
+from xml.sax.saxutils import XMLGenerator
+from colorama import Style
+import atexit
+import codecs
+import os
+import ptpython.repl
+import pty
+import re
+import subprocess
+import sys
+import tempfile
+import time
+import traceback
+import unicodedata
+
+import common
+
+
+def test_script() -> None:
+    exec(os.environ["testScript"])
+
+
+class Driver():
+    def __init__(self, machine_class, log, vm_scripts, keep_vm_state,
+                 configure_python_repl=id,
+                 machine_config_modifier=lambda x: x):
```

Comment:
```suggestion
                 machine_config_modifier=id):
```

Reply:
id isn't the identity function even though it _does_ work in this case

Reply:
How does this "magic" work? I'd never seen that before, but if this spares us declaring lambdas as defaults, then it's great. :+1:

Reply:
`id` gives you the identity (think: address) of an object. Since that doesn't modify the config, it also works in this case, but that's more in the sense that _any_ function that doesn't modify its argument would work here.

Reply:
So it _behaves_ (for all intents and purposes) as a no-op, but _is_ it actually one? (I'd guess not)

Reply:
I took this out in current version, since I suspect it won't be needed anymore. We'll see, though.